library/std/src/sys/unix/args.rs

```diff
@@ -139,9 +141,11 @@ mod imp {
 
     fn clone() -> Vec<OsString> {
         unsafe {
-            let _guard = LOCK.lock();
+            LOCK.lock();
             let argc = ARGC.load(Ordering::Relaxed);
             let argv = ARGV.load(Ordering::Relaxed);
+            LOCK.unlock();
```

Comment:
This PR assumes these `load`s and `store`s will not panic. The docs state "panics if order is `Release`/`Acquire` or `AcqRel`" and the implementation is an intrinsic.